org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/GenericsRegressionTest_1_8.java

```diff
@@ -9173,45 +9173,40 @@ public void testBug525580() {
 			"----------\n" +
 			"2. ERROR in org\\a\\a\\g\\d.java (at line 6)\n" +
 			"	T t = (e) cls.newInstance();\n" +
-			"	      ^^^^^^^^^^^^^^^^^^^^^\n" +
-			"e cannot be resolved to a type\n" +
-			"----------\n" +
-			"3. ERROR in org\\a\\a\\g\\d.java (at line 6)\n" +
-			"	T t = (e) cls.newInstance();\n" +
```

Comment:
This looks like a welcome elimination of an extra message - do you know readily what brought about this ?

Reply:
I think it's this:
* previously `LocalDeclaration.resolveType()` would go into `scope.problemReporter().typeMismatchError()` which marks the entire initialization as bogus (detecting inside problem reporting, that type mismatch is actually caused by not-found type).
* now we detect that also `<T extends e>` refers to the missing type `e`, and since we know (codified since https://bugs.eclipse.org/196200 :)) that an error has already been reported for the type, we don't bother to additionally report incompatibility vs the missing type.

Ergo, the change of behavior would have been caused by also tagging `<T>` as `HasMissingType`, by propagating this flag from `e` to `<T extends e>`.